arbor/label_resolver.cpp

```diff
@@ -0,0 +1,90 @@
+#include <vector>
+
+#include <arbor/assert.hpp>
+#include <arbor/arbexcept.hpp>
+#include <arbor/common_types.hpp>
+
+#include "label_resolver.hpp"
+#include "util/partition.hpp"
+#include "util/rangeutil.hpp"
+
+namespace arb {
+
+cell_labeled_ranges::cell_labeled_ranges(std::vector<cell_gid_type> gid_vec,
+                                         std::vector<cell_size_type> size_vec,
+                                         std::vector<cell_tag_type> label_vec,
+                                         std::vector<lid_range> range_vec):
+    gids(std::move(gid_vec)), sizes(std::move(size_vec)), labels(std::move(label_vec)), ranges(std::move(range_vec))
+{
+    arb_assert(labels.size() == ranges.size());
+    arb_assert(gids.size() == sizes.size());
+};
+
+void cell_labeled_ranges::append(cell_labeled_ranges other) {
+    using util::append;
+    append(gids, other.gids);
+    append(sizes, other.sizes);
+    append(labels, other.labels);
+    append(ranges, other.ranges);
+}
```

Comment:
Right, I tried to add a `util::append` capable of moving the vectors in (using `std::make_move_iterator`) but I ran into some issues there (related to appending `util::transform_view` - data was being moved instead of copied). Granted, I'm not an expert on move semantics yet. Do you think adding this method is worthwhile?

Reply:
If we can get the semantics straight, it would be handy, but it might not be worth the effort (i.e., perhaps just use `std::` functionality.)